src/util/c_types.h

```diff
@@ -327,6 +327,88 @@ inline c_enum_tag_typet &to_c_enum_tag_type(typet &type)
   return static_cast<c_enum_tag_typet &>(type);
 }
 
+class code_with_contract_typet : public code_typet
```

Comment:
I don't think `c_types.h` is the right place for this: there is nothing C-specific about code contracts. It's just that currently only the C front-end has support for these.

Reply:
I'll be happy to promote this once there is any other front-end that offers contracts.

Reply:
src/util otherwise ends up being the union of everything, when it should be for things that are used in most places.

Reply:
Understood. What I'm worried about is the distinction between "this is only meaningful for C" (which is what I thought this file was about) vs "this is only used in the C front-end." But that may well be a matter of taste, we have much more important issues to obsess over ;-)